homeassistant/components/device_tracker/config_entry.py

```diff
@@ -192,7 +192,7 @@ def battery_level(self) -> int | None:
         return None
 
     @property
-    def source_type(self) -> SourceType | str:
+    def source_type(self) -> SourceType | str | None:
```

Comment:
Why should it be optional? That's not intended.

Reply:
It is optional for mwtt and the legacy device tracker.

Reply:
The legacy device tracker doesn't use this class. The mqtt integration should be updated to make it required for config entry device trackers as that's what the design mandates.

Reply:
In that case we will need a default for SourecType, or we will need the change to break the current setup. Is that what we want?

Reply:
That's a problem for the MQTT integration. I don't know why it made it optional, which is incorrect for this case.

Reply:
https://github.com/home-assistant/core/pull/81128